### ISSUE TITLE
internalize_vfunc.cpp failing on AMD HIP and CUDA after merging in 90ba33099cbb

### ISSUE THREAD

**Issue by jsji** [intel/llvm#12517]
**Describe the bug**

https://github.com/intel/llvm/actions/runs/7672516896/job/20913879385
```
FAIL: SYCL :: KernelFusion/internalize_vfunc.cpp (1220 of [18](https://github.com/intel/llvm/actions/runs/7672516896/job/20913879385#step:21:19)50)
******************** TEST 'SYCL :: KernelFusion/internalize_vfunc.cpp' FAILED ********************
Exit Code: -6

Command Output (stdout):
--
# RUN: at line 1
/__w/llvm/llvm/toolchain/bin//clang++  -Xsycl-target-backend=amdgcn-amd-amdhsa --offload-arch=gfx10[31](https://github.com/intel/llvm/actions/runs/7672516896/job/20913879385#step:21:32) -fsycl -fsycl-targets=amdgcn-amd-amdhsa /__w/llvm/llvm/llvm/sycl/test-e2e/KernelFusion/internalize_vfunc.cpp -fsycl-embed-ir -O2 -o /__w/llvm/llvm/build-e2e/KernelFusion/Output/internalize_vfunc.cpp.tmp.out
# executed command: /__w/llvm/llvm/toolchain/bin//clang++ -Xsycl-target-backend=amdgcn-amd-amdhsa --offload-arch=gfx1031 -fsycl -fsycl-targets=amdgcn-amd-amdhsa /__w/llvm/llvm/llvm/sycl/test-e2e/KernelFusion/internalize_vfunc.cpp -fsycl-embed-ir -O2 -o /__w/llvm/llvm/build-e2e/KernelFusion/Output/internalize_vfunc.cpp.tmp.out
# note: command had no output on stdout or stderr
# RUN: at line 2
env ONEAPI_DEVICE_SELECTOR=hip:gpu  /__w/llvm/llvm/build-e2e/KernelFusion/Output/internalize_vfunc.cpp.tmp.out
# executed command: env ONEAPI_DEVICE_SELECTOR=hip:gpu /__w/llvm/llvm/build-e2e/KernelFusion/Output/internalize_vfunc.cpp.tmp.out
# .---command stderr------------
# | internalize_vfunc.cpp.tmp.out: /__w/llvm/llvm/llvm/sycl/test-e2e/KernelFusion/internalize_vfunc.cpp:87: int main(): Assertion `tmp[i] == -1 && "Not internalized"' failed.
# `-----------------------------
# error: command failed with exit status: -6

```
Pass without https://github.com/llvm/llvm-project/commit/90ba33099cbb
https://github.com/intel/llvm/actions/runs/7689888115/job/20953236267

**To Reproduce**

Remove XFAIL and run e2e on AMD

**Comment by jsji** [comment]
@sommerlukas Can you have a look? Thanks.

**Comment by sommerlukas** [comment]
@jsji Thanks for reporting this, I've filed an internal ticket for this and we will investigate.

**Comment by sommerlukas** [comment]
@jsji Closing this, as the issue should be fixed with #12557. We have removed the `XFAIL` from the test again in #12557.

**Comment by jsji** [comment]
Thank you @sommerlukas !
